src/prefect/engine.py

```diff
@@ -618,7 +618,14 @@ async def orchestrate_flow_run(
         )
 
     if flow.flow_run_name:
-        flow_run_name = flow.flow_run_name.format(**parameters)
+        if callable(flow.flow_run_name):
+            flow_run_name = flow.flow_run_name(parameters)
```

Comment:
**WARNING**: The parameters is sent directly to the `flow_run_name` function which could leave it vulnerable to mutation. We could try to send a shallow copy of the parameters dict, but still the individual args themselves could be mutated ( similar to how it's vulnerable in `str.format(**parameters)` ).

I'll leave the call on what to do here to the Prefect team.

Reply:
I think that's okay with me for now 👍